### ISSUE TITLE
Data.IntMap.mapAccumWithKey traverses positive keys before negative keys

### ISSUE THREAD

**Issue by sjakobi** [haskell/containers#689]
```
> mapAccumWithKey (\acc k v -> (acc ++ show k ++ show v, v)) "" (Data.IntMap.fromList [(-1, "a"), (2, "b"),(-3, "c")])
("2\"b\"-3\"c\"-1\"a\"",fromList [(-3,"c"),(-1,"a"),(2,"b")])
```

https://github.com/haskell/containers/issues/579 was a similar bug.

**Comment by sjakobi** [comment]
Both the strict and the lazy versions need to be fixed here.

**Comment by Vlix** [comment]
I've adjusted `mapAccumWithKey` & `mapAccumRWithKey` both lazy and strict in #688.
And adjusted tests to also include negative numbers.

**Comment by sjakobi** [comment]
Cheers, @Vlix! :+1:

**Comment by sjakobi** [comment]
Fixed via #688.
